atomix/cluster/src/main/java/io/atomix/raft/cluster/impl/RaftMemberContext.java

```diff
@@ -96,8 +100,9 @@ public void resetState(final RaftLog log) {
   public boolean canAppend() {
     return appending == 0
         || (appendSucceeded
-            && appending < MAX_APPENDS
-            && System.currentTimeMillis() - (timeStats.getMean() / MAX_APPENDS) >= appendTime);
+            && appending < maxAppendsPerMember
+            && System.currentTimeMillis() - (timeStats.getMean() / maxAppendsPerMember)
```

Comment:
Random question, I know this has nothing to do with your PR, but...I found this kind of hard to read. So we can append if the last append succeeded, the current append count is less than max appends, and the time since the last append is greater than or equal to the mean of the 8 observed last append latencies divided by the max number of appends? Did I read that right?

Reply:
Could be I not really get the last part tbh

Reply:
So yes it seems it calculates the latencies by hand and tries to only send a new append when we are over the avg latency - weird

Reply:
wasnt aware of that

Reply:
Could we just send more append and the follower could also just ack the last one? then we know it also received the previous?

Reply:
So it sort of tries to do flow control by not sending appends faster than what the other node has managed to acknowledge, I guess. Wouldn't this potentially calculate though a high average? Let's say the follower has a slow spike, if we only send based on the rolling average, would that really go lower over time? Probably not to the optimal value

Reply:
Yes, I think in the long run we want to do an async pipeline where the follower also acknowledges the last write it flushed (with async flush, not sync) anyway.

Reply:
It only evaluates the last 8 values calcs the mean of that and then divides that by 2 to get the avg ? Hm Idk we probably need to experiment on this thing or just remove it ^^